bulkExport/uploadGlueScriptsToS3.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import * as AWS from 'aws-sdk';
+import axios from 'axios';
+
+const sendCfnResponse = async (event, status: 'SUCCESS' | 'FAILED', error?: Error) => {
+    const responseBody = JSON.stringify({
+        Status: status,
+        Reason: error?.message,
+        // The value of PhysicalResourceId doesn't really matter in this case.
+        // It just needs to be the same string on all responses to indicate that it is the same resource.
+        PhysicalResourceId: 'glueScripts',
+        StackId: event.StackId,
+        RequestId: event.RequestId,
+        LogicalResourceId: event.LogicalResourceId,
+    });
+    console.log(`Sending response to CFN: ${responseBody}`);
+    await axios.put(event.ResponseURL, responseBody);
+};
+/**
+ * Custom resource lambda handler that uploads a specific file to s3.
+ * Custom resource spec: See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html
+ * @param event Custom resource request event. See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requests.html
+ */
+exports.handler = async event => {
```

Comment:
So this lambda is kicked off by the CFN custom resource deployment, and we ensure that the lambda is invoked every deployment by having that RandomValue parameter inside the custom resource.

Does the deployment order matter? ie CustomResource deployed first _then_ this lambda?

Reply:
Custom resources are only updated when their parameters change(similar to regular CFN resources). The random value forces the update to happen on all deployments (there's no harm in re-uploading the same file)

A custom resource is just like any other CFN resource. CFN determines the deployment order from the dependency graph. The custom resource implicitly depends on the lambda function since it references it at:
```
ServiceToken: !GetAtt UploadGlueScriptsLambdaFunction.Arn
```

Reply:
Got it thanks - should the AWS::Glue::Job resource depend on the custom resource (to help ensure that the file has been uploaded to S3)?

Reply:
I added the `DependsOn`.

Most of the time it doesn't matter since the python script is just needed at run time, not at deploy time. But it does prevent the very rare case where a customer manages to start a bulk export request BEFORE their first CREATE stack operation completes and the job fails to find the script in s3.